### PR TITLE
[TASK] Revise "Dependency injection configuration" section

### PR DESCRIPTION
**Pull request by brotkrueml**
Also removes the "WIP" before it as it is already comprehensive.

Releases: main, 11.5

### CODE REVIEW

File: Documentation/ApiOverview/DependencyInjection/Index.rst [hunk 1]
```diff
@@ -471,110 +471,113 @@ Configure dependency injection in extensions
...
+Alternatively, :file:`Configuration/Services.php` can also be used.
 A basic :file:`Services.yaml` file of an extension looks like the following.
 
-*Note:*
-Whenever service configuration or class dependencies change, the Core cache needs
-to be flushed in the Install Tool to rebuild the compiled Symfony container.
-Flushing all caches from the cache clear menu does not flush the compiled Symfony container.
+..  note::
+    Whenever the service configuration or class dependencies change, the Core
+    cache must be flushed in the Install Tool or via the :ref:`CLI <cli-mode>`
```
Comment (linawolf) [inline Documentation/ApiOverview/DependencyInjection/Index.rst:479]:
```suggestion
    cache must be flushed in the :guilabel:`Admin Tools` or via the :ref:`CLI <cli-mode>`
```

File: Documentation/ApiOverview/DependencyInjection/Index.rst [hunk 2]
```diff
@@ -471,110 +471,113 @@ Configure dependency injection in extensions
...
+    command `cache:flush` to rebuild the compiled Symfony container. Flushing
+    all caches from the "Clear cache" menu does not flush the compiled Symfony
```
Comment (linawolf) [inline Documentation/ApiOverview/DependencyInjection/Index.rst:481]:
```suggestion
    all caches from the :guilabel:`Clear cache` menu does not flush the compiled Symfony
```

File: Documentation/ApiOverview/DependencyInjection/Index.rst [hunk 3]
```diff
@@ -471,110 +471,113 @@ Configure dependency injection in extensions
...
+    (:yaml:`shared: true`).
 
 Model exclusion
-   The path exclusion :yaml:`exclude: '../Classes/Domain/Model/*'` excludes
-   your models from the DI container, which means that you cannot inject them nor inject
-   dependencies into them.
-   Models are not services and should therefore not require dependency injection.
-   Also, these objects are created by the Extbase persistence layer which does not support the DI container.
+    The path exclusion :yaml:`exclude: '../Classes/Domain/Model/*'` excludes
+    your models from the DI container, which means you cannot inject them
```
Comment (linawolf) [inline Documentation/ApiOverview/DependencyInjection/Index.rst:518]:
```suggestion
    your models from the dependency injection container, which means you cannot inject them
```

File: Documentation/ApiOverview/DependencyInjection/Index.rst [hunk 4]
```diff
@@ -584,97 +587,104 @@ These classes include singletons, because they need to be shared with code that
...
 
-This means every class that is retrieved directly using :php:`GeneralUtility::makeInstance()` *and* needs dependency injection, *must* be marked as public.
-Any other class which needs dependency injection and is retrieved by dependency injection
-itself can be private.
-Instances of :php:`\TYPO3\CMS\Core\SingletonInterface` and Extbase controllers are
-automatically marked as public because they are retrieved using :php:`GeneralUtility::makeInstance()`.
-More examples for classes which must be marked as public:
+This means every class that is directly retrieved using
+:php:`GeneralUtility::makeInstance()` *and* requires dependency injection
+*must* be marked as public. Any other class which requires dependency injection
```
Comment (linawolf) [inline Documentation/ApiOverview/DependencyInjection/Index.rst:607]:
```suggestion
**must** be marked as public. Any other class which requires dependency injection
```
I would make this bold

File: Documentation/ApiOverview/DependencyInjection/Index.rst (hunk 2 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/DependencyInjection/Index.rst:481]: Yes thought also about it. But there is no menu labeled "Clear cache" just the flash icon, so guilabel is IMHO not right.

Comment (linawolf) [reply Documentation/ApiOverview/DependencyInjection/Index.rst:481]: But the icon has the title and therefore does have the name?

**Review by linawolf — APPROVED** [review]

**Pull request merged**